pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go

```diff
@@ -0,0 +1,510 @@
+	client.Client
+	scheme   *runtime.Scheme
+	recorder record.EventRecorder
+	mapper   meta.RESTMapper
+
+	configration config.LoadBalancerSetControllerConfiguration
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(c *appconfig.CompletedConfig, mgr manager.Manager) *ReconcileLoadBalancerSet {
+	return &ReconcileLoadBalancerSet{
+		Client:       mgr.GetClient(),
+		scheme:       mgr.GetScheme(),
+		mapper:       mgr.GetRESTMapper(),
+		recorder:     mgr.GetEventRecorderFor(names.LoadBalancerSetController),
+		configration: c.ComponentConfig.LoadBalancerSetController,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, cfg *appconfig.CompletedConfig, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(names.LoadBalancerSetController, mgr, controller.Options{
+		Reconciler: r, MaxConcurrentReconciles: int(cfg.ComponentConfig.LoadBalancerSetController.ConcurrentLoadBalancerSetWorkers),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to PoolService
+	err = c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForObject{}, NewServicePredicated())
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &netv1alpha1.PoolService{}}, NewPoolServiceEventHandler(), NewPoolServicePredicated())
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &v1beta1.NodePool{}}, NewNodePoolEventHandler(mgr.GetClient()), NewNodePoolPredicated())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=network.openyurt.io,resources=poolservices,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=network.openyurt.io,resources=poolservices/status,verbs=get;update;patch
+
+// Reconcile reads that state of the cluster for a PoolService object and makes changes based on the state read
+// and what is in the PoolService.Spec
+func (r *ReconcileLoadBalancerSet) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	// Note !!!!!!!!!!
+	// We strongly recommend use Format() to  encapsulation because Format() can print logs by module
+	// @kadisi
+	klog.Infof(Format("Reconcile PoolService %s/%s", request.Namespace, request.Name))
+
+	service := &corev1.Service{}
+	err := r.Get(context.TODO(), request.NamespacedName, service)
+	if apierrors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get service %s", request.String())
+	}
+
+	copySvc := service.DeepCopy()
+
+	if shouldDeleteAllPoolServices(copySvc) {
+		if err := r.deleteAllPoolServices(copySvc); err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to clean pool services")
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.reconcileService(copySvc); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to apply pool services")
```

Comment:
failed to apply pool services --> failed to sync service(%s/%s), copySvc.Namespace, copySvc.Name

Reply:
done